scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -270,37 +284,31 @@ def _load_modules_generic(self, modules_to_load):
         ###################################################
         # Parse the output to set the os.environ dictionary
         ###################################################
-        newenv = {}
-        dolater = []
+        newenv = OrderedDict()
+        lastkey = None
         for line in output.splitlines():
-            if line.find('$')>0:
-                dolater.append(line)
-                continue
-
-            m=re.match(r'^(\S+)=(\S+)\s*;*\s*$',line)
-            if m:
-                key = m.groups()[0]
-                val = m.groups()[1]
+            if "=" in line:
+                key, val = line.split("=", 1)
                 newenv[key] = val
+                lastkey = key
+            else:
+                newenv[lastkey] += "\n" + line
 
-        # Now that initial newenv has been set, resolve variables
-        for line in dolater:
-            m=re.match(r'^(\S+)=(\S+)\s*;*\s*$',line)
-            if m:
-                key = m.groups()[0]
-                valunresolved = m.groups()[1]
-                val = string.Template(valunresolved).safe_substitute(newenv)
-                expect(val is not None,
-                       'string value {} unable to be resolved'.format(valunresolved))
-                newenv[key] = val
+        # resolve variables
+        for key, val in newenv.items():
+            newenv[key] = string.Template(val).safe_substitute(newenv)
 
         # Set environment with new or updated values
         for key in newenv:
```

Comment:
There was a bug in the old code, environment variables were never being deleted. Made no sense to loop over newenv and then check if key not in newenv.